lang/gapi.go

```diff
@@ -67,10 +68,26 @@ func (obj *GAPI) Cli(c *cli.Context, fs resources.Fs) (*gapi.Deploy, error) {
 			return nil, fmt.Errorf("input code is empty")
 		}
 
+		// validate syntax before deploying
+		filecontent, err := ioutil.ReadFile(s)
+		if err != nil {
+			return nil, errwrap.Wrapf(err, "can't read code from file `%s`", s)
+		}
+		code := strings.NewReader(string(filecontent))
+		obj := &Lang{
+			Input: code,
+			Logf: func(format string, v ...interface{}) {
+				log.Printf(Name+": funcs: "+format, v...)
+			},
+		}
+		err = obj.Validate()
+		if err != nil {
+			return nil, errwrap.Wrapf(err, s)
+		}
+
 		// read through this local path, and store it in our file system
 		// since our deploy should work anywhere in the cluster, let the
 		// engine ensure that this file system is replicated everywhere!
-
```

Comment:
why is this line missing?
if there's a reason, okay, but it sends the message you don't review your diffs before submitting.
the goal should be for me to see your patch and say "perfect, merged" :)

Reply:
must have slipped past

Reply:
no worries